### PR TITLE
[WW-5117] Evaluates dynamic attributes - cherrypick

### PR DESCRIPTION
**Pull request by lukaszlenart**
Cherry picks #524

### CODE REVIEW

File: core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java [hunk 1]
```diff
@@ -346,7 +346,7 @@ public void testSimple_recursionTest() throws Exception {
         tag.setName("myname");
         tag.setValue("%{foo}");
         tag.setSize("10");
-        tag.setDynamicAttribute(null, "anotherAttr", "%{foo}");
+        tag.setDynamicAttribute(null, "anotherAttr", "another_%{foo}");
```
Comment (yasserzamani) [inline core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java:349]: That's intentional by me to start value with %{ to always have a monitor and test for double evaluation. This change eliminate that I think. Why had you to change it? to pass that test?

File: core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java [hunk 2]
```diff
@@ -372,7 +372,7 @@ public void testSimple_recursionTest_clearTagStateSet() throws Exception {
...
```
Comment (yasserzamani) [inline core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java:375]: (same here)

File: core/src/test/resources/org/apache/struts2/views/jsp/ui/Textfield-5.txt [hunk 3]
```diff
@@ -1,4 +1,4 @@
 <tr>
     <td class="tdLabel"><label for="myname" class="label">mylabel:</label></td>
-    <td class="tdInput"><input type="text" name="myname" size="10" value="%{1+1}" id="myname" anotherAttr="%{1+1}"/></td>
+    <td class="tdInput"><input type="text" name="myname" size="10" value="%{1+1}" id="myname" anotherAttr="another_%{1+1}"/></td>
```
Comment (yasserzamani) [inline core/src/test/resources/org/apache/struts2/views/jsp/ui/Textfield-5.txt:3]: (same here)

**Review by yasserzamani — CHANGES_REQUESTED** [review]

File: core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java (hunk 1 above)
Comment (yasserzamani) [reply core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java:349]: Thanks for clarification! Just wanted to note and clarify it. So we can go ahead :) 

p.s. however I think it's a good practice to not touch current tests and just add tests, just for more confident that changes aren't breaking previous assertions. e.g. in this case I think we can simply add a new attr **anotherAttr2**:
```tag.setDynamicAttribute(null, "anotherAttr2", "another_%{foo}");```
and add expected attr to the current expected file. I see your point that dynAttr are normally `static_text_%{variable}` but this test is just a special not normal case test which assumes that attacker had been successful to inject such value.

**Review by yasserzamani — APPROVED** [review]

File: core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java (hunk 1 above)
Comment (lukaszlenart) [reply core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java:349]: Sure, I can do it, no problem :)

Comment (lukaszlenart) [reply core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java]: Done!

Comment (yasserzamani) [reply core/src/test/java/org/apache/struts2/views/jsp/ui/TextfieldTest.java]: thanks so much! LGTM :+1: will wait a few days and merge

**Pull request merged**